rep-2004.rst

```diff
@@ -0,0 +1,689 @@
+   #. Must have confirmation of contributor origin (e.g. `DCO  <https://developercertificate.org/>`_, CLA, etc.)
+   #. Must have peer review policy for all change requests (e.g. require one or more reviewers)
+   #. Must have Continuous Integration (CI) policy for all change requests
+   #. Must have documentation policy for all change requests
+
+#. **Documentation:**
+
+   #. Must have documentation for each "feature" (e.g. for ``rclcpp``: create a node, publish a message, spin, etc.)
+   #. Must have documentation for each item in the public API (e.g. functions, classes, etc.)
+   #. Must have a declared license or set of licenses
+   #. Must have a copyright statement in each source file
+   #. Must have a "quality declaration" document, which declares the quality level and justifies how the package meets each of the requirements
+
+      #. *Must have a section in the repository's ``README`` which contains the "quality declaration" or links to it*
+      #. *Must register with a centralized list of 'Level N' packages, if one exists, to allow for peer review of the claim*
+
+#. **Testing:**
+
+   #. Must have system tests which cover all items in the "feature" documentation
+   #. Must have system, integration, and/or unit tests which cover all of the public API
+   #. Code Coverage:
+
+      #. *Must have code coverage tracking for the package*
+      #. *Must have and enforce a code coverage policy for new changes*
+
+   #. Performance:
+
+      #. *Must have performance tests (exceptions allowed if they don't make sense to have)*
+      #. *Must have a performance regression policy (i.e. blocking either changes or releases on unexpected performance regressions)*
+
+   #. Linters and Static Analysis:
+
+      #. *Must have a code style and enforce it*
+      #. *Must use static analysis tools where applicable*
+
+#. **Dependencies:**
+
+   #. Must not have direct runtime "ROS" dependencies which are not at the same level as the package in question ('Level N'), but...
+   #. May have optional direct runtime "ROS" dependencies which are not 'Level N', e.g. tracing or debugging features that can be disabled
+   #. Must have justification for why each direct runtime "non-ROS" dependency is equivalent to a 'Level N' package in terms of quality
+
+#. **Platform Support:**
+
+   #. Must support all target platforms for the package's ecosystem.
+
+      * For ROS 2 this means supporting all tier 1 platforms, as defined in `REP-2000 <https://www.ros.org/reps/rep-2000.html#support-tiers>`_
+
+If the above points are satisfied then a package can be considered 'Level 1'.
+Refer to the detailed description of the requirements below the chart for more information.
+
+Quality Level Comparison Chart
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The chart below compares Quality Levels 1 through 5 relative to the 'Level 1' requirements' numbering scheme above.
+
+✓ = required
+
+● = recommended
+
+.. list-table:: Quality Levels
```

Comment:
The table is great! I'm not sure if this has already been discussed, but to reduce repetition, the items could be described only once, and the table serve as the reference for all levels. As opposed to the items being repeated for every level.

Reply:
We did have this conversation and I think there were a couple (soft) reasons for having both:
Jumping back and forth to the level 1 description will be annoying, so it's better to give people the option to read through the requirements for each level since it's easier. The text lists can stand alone as a resource and the chart can be a quick reference just for comparison's sake, not content.

Neither of us felt really strongly about either option though. It is pretty repetitive

Reply:
Ok, I can see that. I don't feel strongly about this, but thanks for the clarification.